src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -303,8 +303,8 @@ def _run_ceph_volume_command(self, host: str,
             error_ok=True)
         return out, err, code
 
-    def get_osdspec_affinity(self, osd_id: str) -> str:
-        return self.mgr.get('osd_metadata').get(osd_id, {}).get('osdspec_affinity', '')
+    def get_osd_metadata(self) -> dict:
```

Comment:
might want to update `src/pybind/mgr/cephadm/tests/fixtures.py` accordingly. otherwise we have errors like
```
    @pytest.fixture()
    def cephadm_module():
>       with with_cephadm_module({}) as m:

cephadm/tests/conftest.py:11: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
cephadm/tests/fixtures.py:42: in with_cephadm_module
    with mock.patch("cephadm.module.CephadmOrchestrator.get_ceph_option", get_ceph_option),\
/usr/lib/python3.8/unittest/mock.py:1393: in __enter__
    original, local = self.get_original()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <unittest.mock._patch object at 0x7fd94fa3d970>

    def get_original(self):
        target = self.getter()
        name = self.attribute
    
        original = DEFAULT
        local = False
    
        try:
            original = target.__dict__[name]
        except (AttributeError, KeyError):
            original = getattr(target, name, DEFAULT)
        else:
            local = True
    
        if name in _builtins and isinstance(target, ModuleType):
            self.create = True
    
        if not self.create and original is DEFAULT:
>           raise AttributeError(
                "%s does not have the attribute %r" % (target, name)
            )
E           AttributeError: <class 'cephadm.services.osd.OSDService'> does not have the attribute 'get_osdspec_affinity'
```

Reply:
@strenuous-life ping?